integration_test/tests/power_bi_tests.rs

```diff
@@ -9,11 +9,54 @@ mod integration {
         AttrConnectionPooling, AttrOdbcVersion, ConnectionAttribute, DriverConnectOption,
         EnvironmentAttribute, HDbc, HEnv, Handle, HandleType, InfoType, Pointer, SQLAllocHandle,
         SQLDriverConnectW, SQLFreeHandle, SQLGetInfoW, SQLSetConnectAttrW, SQLSetEnvAttr, SmallInt,
-        SqlReturn,
+        SqlReturn, USmallInt,
     };
     use std::ptr::null_mut;
     use std::slice;
 
+    macro_rules! test_get_info {
+        ($conn_handle:expr, $env_handle: expr, $info_type: expr , $actual_value_modifier: ident) => {{
+            let conn_handle = $conn_handle;
+            let env_handle = $env_handle;
+            let info_type = $info_type;
+            let actual_value_modifier = $actual_value_modifier;
```

Comment:
I see what you are doing here. Since the buffer is "polymorphic" and the type of data stored inside is different, you want to branch out to retrieve the right kind.
I am wondering if it would not be cleaner and more straightforward to have a struct and impl `into` to handle converting buffer content to Rust data.

Something like this might work. 
```
pub struct OutputBuffer<T> {
    pub output_buffer: Pointer,
    pub mut data_length: i32,
    pub buffer_length: i32,
}

impl Into<String> for OutputBuffer<WChar> {
    fn into(self) -> String {
        unsafe {
            String::from_utf16_lossy(slice::from_raw_parts(
                self.output_buffer as *const WChar,
                *self.data_length as usize,
            ))
        }
    }
}

impl Into<u16> for OutputBuffer<USmallInt> {
    fn into(self) -> u16 {
        unsafe { *(self.output_buffer as *mut u16) }
    }
}
....

```

I am not sure if this would make things easier or more bulky though, so if you find out that it's not worth it, that's Ok. I am just thinking that having this would be useful for reading the content of any output buffer, anywhere in the tests.

Reply:
I made changes based on this suggestion but was hitting an error where it seemed that which Into to use could not be inferred by Rust.  I suspect it is because the USmallint and WChar types are both u16.
```
note: multiple `impl`s satisfying `OutputBuffer<u16>: Into<_>` found
   --> integration_test\tests\power_bi_tests.rs:70:5
    |
70  |     impl Into<String> for OutputBuffer<WChar> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
81  |     impl Into<u16> for OutputBuffer<USmallInt> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
So to determine which type, I added a `DataType` Enum and specify that when using the macro.